### PR TITLE
[Datafactory] Add configure repo api swagger

### PR DESCRIPTION
**Pull request by frankycrm14**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by hvermis — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (frankycrm14) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: As we talked offline about this, I'll explore other options, and ask feedback on this from the ARM tema.

Comment (marstr) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: I doubt I need to tell you, @ravbhatnagar, but make sure to take a look at this comment thread. :)

**Review by marstr — COMMENTED** [review]
I've verified that non of the CI failures regarding the files in this PR are introduced by these changes. Other than the already noted path concerns, this PR looks good from a code-generation perspective.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (ravbhatnagar) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: Just replied on email, IMO this should be modeled as an action on /factories/{factoryName} as mentioned by @hvermis

Comment (ravbhatnagar) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: Also, this returns a factory resource, Is it fine where someone who just has permission to this POST action can read the resource even though he may not have /read or /write permission. I am sure this API is not creating a factory resource. Is it changing the shape of the factory resource?

Comment (frankycrm14) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: this changes some properties of the factory, to call this api you will need to have permissions for this post call and also write permissions to the factory

**Review by hvermis — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: This is the full ARM resource ID for the factory right? Since your API is under factory resource now, you shouldn't need this, since all information will be in your path.

**Review by hvermis — CHANGES_REQUESTED** [review]
Make sure to address the comments on your last commit too.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (frankycrm14) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: I need this property for the linkedActionCheck

Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: You can construct it since you have the factory name, rg name, sub id in your headers when the API gets called. We shouldn't require the client to send this extra field if we already have all the information.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/examples/Factories_ConfigureRepo.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/examples/Factories_ConfigureRepo.json]: See, this Id gets returned. So in the code you might even have this as one of the parameters passed to your API.

**Review by hvermis — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/examples/Factories_ConfigureRepo.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/examples/Factories_ConfigureRepo.json]: Talked offline, the resource ID is for the cases when the call goes through the desired action "Microsoft.DataFactory/factories/configureRepo/action" and ARM needs it to validate the permissions. Recalling my comment.

**Review by hvermis — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
Signing off as the feedback given over email has been addressed.

**Comment by marstr** [comment]
Are these changes deployed to the public? Are you ready for me to merge?

**Comment by frankycrm14** [comment]
I'm testing the api to verify that the approach we took actually works.

**Comment by marstr** [comment]
Still working on this PR, @frankycrm14? What's its status?

**Comment by frankycrm14** [comment]
It’s done, but I’m waiting for deployment on the backend, I was told that’s the way to go, not to change swagger until changes are deployed in backend’s prod.
It’s been taking some time to deploy on the backend side for some issues that were found, but hopefully it will be done soon.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (marstr) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: I'd ref this definition instead, in order to get client-side validation:
https://github.com/Azure/azure-rest-api-specs/pull/2866/files#diff-bf6d272fbfeeed92d1fc24a2bb8706a8R3781

**Review by marstr — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json
Comment (marstr) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json]: Chatted offline with @frankycrm14 about this, it looks like the defintion I mentioned is in the parameters section. For that reason, the real fix would be to refactor most of this into the definitions section, and reference that from the parameter and anywhere else it's used. However, that would only buy us a little client-side validation, and wouldn't be a breaking change to fix in the future. For those reasons, I'm not going to block this PR on this kinda comment.

All that remains before I merge is ensuring that none of the linter errors being reported were introduced by this PR.

**Review by marstr — APPROVED** [review]
To the best of my knowledge, I believe the CI errors here predate this PR.

**Pull request merged**
